ghost/core/test/unit/server/services/members/members-api/controllers/RouterController.test.js

```diff
@@ -1097,4 +1097,313 @@ describe('RouterController', function () {
+            VALID_SECRET: 'a'.repeat(128),
+            SHORT_SECRET: 'a'.repeat(64),
+            SITE_URL: 'http://example.com',
+            MEMBERS_URL: 'http://example.com/members/',
+            ERROR_MESSAGES: {
+                BAD_REQUEST: 'Bad Request.',
+                OTC_REQUIRED: 'otc and otcRef are required',
+                INVALID_FORMAT: 'Invalid verification code',
+                FAILED_VERIFY: 'Failed to verify code, please try again',
+                INVALID_CODE: 'Invalid verification code'
+            },
+            SUCCESS_MESSAGES: {
+                OTC_VERIFICATION_SUCCESSFUL: 'OTC verification successful'
+            }
+        };
+
+        beforeEach(function () {
+            mockMagicLinkService = {
+                tokenProvider: {
+                    verifyOTC: sinon.stub(),
+                    getTokenByRef: sinon.stub()
+                },
+                getSigninURL: sinon.stub().returns(OTC_TEST_CONSTANTS.MEMBERS_URL)
+            };
+
+            mockSettingsCache = {
+                get: sinon.stub()
+            };
+
+            mockUrlUtils = {
+                urlFor: sinon.stub().returns(OTC_TEST_CONSTANTS.MEMBERS_URL),
+                getSiteUrl: sinon.stub().returns(OTC_TEST_CONSTANTS.SITE_URL)
+            };
+
+            routerController = new RouterController({
+                offersAPI,
+                paymentsService,
+                tiersService,
+                memberRepository: {},
+                StripePrice: {},
+                allowSelfSignup: () => true,
+                magicLinkService: mockMagicLinkService,
+                stripeAPIService,
+                tokenService: {},
+                memberAttributionService: {},
+                sendEmailWithMagicLink: sinon.stub(),
+                labsService,
+                newslettersService: {},
+                sentry: undefined,
+                settingsCache: mockSettingsCache,
+                urlUtils: mockUrlUtils
+            });
+
+            req = {
+                body: {
+                    otc: OTC_TEST_CONSTANTS.VALID_OTC,
+                    otcRef: OTC_TEST_CONSTANTS.TOKEN_ID
+                }
+            };
+
+            res = {
+                writeHead: sinon.stub(),
+                end: sinon.stub()
+            };
+        });
+
+        describe('input validation', function () {
+            it('should throw BadRequestError when otc is missing', async function () {
+                req.body.otc = undefined;
+
+                try {
+                    await routerController.verifyOTC(req, res);
+                    assert.fail('Expected BadRequestError to be thrown');
+                } catch (error) {
+                    assert(error instanceof errors.BadRequestError);
+                    assert.equal(error.message, OTC_TEST_CONSTANTS.ERROR_MESSAGES.BAD_REQUEST);
+                    assert.equal(error.context, OTC_TEST_CONSTANTS.ERROR_MESSAGES.OTC_REQUIRED);
```

Comment:
We shouldn't ever be testing exact error message content unless they contain dynamic data. If we're expecting a specific error of type `x` then the error should have a unique `code` set on it and tested against.